PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessClusterDestoryDialog.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.sparkserverless.serverexplore.ui;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.microsoft.azure.hdinsight.common.mvc.SettableControl;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessCluster;
+import com.microsoft.azure.sparkserverless.serverexplore.SparkServerlessClusterDestoryCtrlProvider;
+import com.microsoft.azure.sparkserverless.serverexplore.SparkServerlessClusterDestoryModel;
+import com.microsoft.azure.sparkserverless.serverexplore.sparkserverlessnode.SparkServerlessClusterNode;
+import com.microsoft.intellij.rxjava.IdeaSchedulers;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.event.*;
+
+public class SparkServerlessClusterDestoryDialog extends DialogWrapper
+        implements SettableControl<SparkServerlessClusterDestoryModel> {
+
+    @NotNull
+    private SparkServerlessClusterDestoryCtrlProvider ctrlProvider;
+
+    @NotNull
+    private SparkServerlessClusterNode clusterNode;
+
+    @NotNull
+    private AzureSparkServerlessCluster cluster;
+
+    private JPanel destroyDialogPanel;
+    private JTextField clusterNameField;
+    private JTextField errorMessageField;
+
+    public SparkServerlessClusterDestoryDialog(@NotNull SparkServerlessClusterNode clusterNode,
+                                               @NotNull AzureSparkServerlessCluster cluster) {
+        super((Project) clusterNode.getProject(), true);
```

Comment:
Why we need the conversion of `(Project)`?

Reply:
Because `getProject` method in class `Node` returns `Object` type. You can check the definition of `getProject` method below.

```java
    public Object getProject() {
        // delegate to parent node if there's one else return null
        if (parent != null) {
            return parent.getProject();
        }

        return null;
    }
```